From edbc08670ede80b292ca4642b203cb602ca2bf7c Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Fri, 7 Mar 2025 07:19:10 +0100
Subject: [PATCH] xilinx: Allow alternative boot strategies in zynq-common.h

Allow config headers that include zynq-common.h to provide their own
(distro) boot strategies. This is implemented by skipping the section
when BOOT_ENV has already been defined.

Signed-off-by: Mike Looijmans <mike.looijmans@topic.nl>
Upstream-Status: Submitted
---
 include/configs/zynq-common.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/configs/zynq-common.h b/include/configs/zynq-common.h
index 553bb1b45b..5763eff65a 100644
--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -48,6 +48,9 @@
 #define BOOTENV
 #else
 
+/* Only use this section if no BOOTENV has been configured yet */
+#ifndef BOOTENV
+
 #ifdef CONFIG_CMD_MMC
 #define BOOT_TARGET_DEVICES_MMC(func) func(MMC, mmc, 0) func(MMC, mmc, 1)
 #else
@@ -167,6 +170,7 @@
 	BOOT_TARGET_DEVICES_DHCP(func)
 
 #include <config_distro_bootcmd.h>
+#endif /* BOOTENV */
 #endif /* CONFIG_SPL_BUILD */
 
 /* Default environment */
-- 
2.43.0

